free-4.md

```diff
@@ -0,0 +1,80 @@
+
+### Operation (semantics)
+_What might happen when a program runs? How does a program interact with the
+user? What kinds of errors might occur, and how might they be communicated to
+the user?_
+
+When a program runs, the user is given a truncated list of recipes they can make given their ingredients and potentially their categorization details. Some errors that might occur include no recipe existing in the intial database, improperly filled fields, misspelled ingredients, among others. It feels most errors will probably come form the use of natural language in the DSL.
+
+### Expressiveness
+_What should be easy to do in this language? What should be possible, but
+difficult? What should be impossible or very difficult?_
+
+The scope of this language is pretty limited, so it should be easy to find a recipe to cook for a meal, and difficult/very difficult/impossible to do much of anythign else.
+
+### Related work
+_Are there any other DSLs in this domain? If not, describe how you know there
+aren't and conjecture why not. If so, describe them and provide links. How well 
+do they address the need? Are there any particularly admirable qualities of the
+language? Are there parts of the language you think could be improved?_
+
+I didn't realize this before hand, but a quick Google-ing of the idea and [this](http://www.supercook.com/) comes up. It is similar, and seems to execute the idea rather well. I'm actually surprised this is not used more often as it seems like a pretty useful idea. It has a good database of existing recipes as well as user submitte recipes. The interface is a little slow and unresponsive though, which would be one area of improvement.
+
+## The Project
+This section examines whether the idea makes for a good CS 111 project.
+
+
+### Suitability
+_If someone were to work on this project, what percentage of their time would be
+spent directly engaging in the **language** aspects of this project (e.g.,
+making language design decisions), as opposed to "systems" aspects of the
+project (e.g., implementing a complicated semantics that doesn't require a lot
+of language design)?_
+
+This project seems to be more about the "systems" aspect, such as finding an efficient way to store all these recipes and ingredients, finding a good way to fetch these recipes given a list of ingredients, amongst other implementation details. The language aspect seems to pale in comparison to the aforementioned problems.
+
+### Scope
+_How big an idea is this? How ambitious is this project?_
+
+This seems like a potentially very large idea. Implementing the features as described above would probably be impossible to complete well in half a semester. It seems also difficult to implement the features mentioned above even with a team of people.
+
```

Comment:
It seems possible if you keep it at a small enough scale with limited functionality, but I agree that the majority of the project would focus on the recipe database. It would need to be organized by ingredients, tags, and more. This might get overwhelming very quickly!